app/pages/neworg/routes.test.js

```diff
@@ -0,0 +1,254 @@
+    .mockImplementation(() => Promise.resolve({})),
+  postData: jest.fn()
+    .mockImplementation(() => Promise.resolve({ success: true })),
+  putData: jest.fn()
+    .mockImplementation(() => Promise.resolve({ success: true })),
+}));
+
+const mockLogoutMethod = jest.fn().mockImplementation(() => Promise.resolve({}));
+
+const mockAuthorisedJwtPayload = JSON.stringify({
+  id: '88421113', name: 'Cool Dude', organisation: 'view',
+});
+
+const mockAuthorisedCookie = `fakeToken=${mockAuthorisedJwtPayload}`;
+
+const setUpFakeApp = () => {
+  const authProvider = new FakeAuthProvider(mockLogoutMethod);
+  const app = new App(authProvider).createApp();
+  app.use('/', routes(authProvider));
+  return app;
+};
+
+const checkAuthorisedRouteNotLoggedIn = path => (
+  request(setUpFakeApp())
+    .get(path)
+    .expect(302)
+    .then((res) => {
+      expect(res.redirect).toEqual(true);
+      expect(res.headers.location).toEqual('http://identity-server/login');
+    }));
+
+const checkAuthorisedRouteWithoutClaim = (path) => {
+  const mockUnauthorisedJwtPayload = JSON.stringify({
+    id: '88421113', name: 'Cool Dude',
+  });
+  const mockUnauthorisedCookie = `fakeToken=${mockUnauthorisedJwtPayload}`;
+
+  request(setUpFakeApp())
+    .get(path)
+    .set('Cookie', [mockUnauthorisedCookie])
+    .expect(200)
+    .then((res) => {
+      expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
+      expect(res.text.includes('Not authorised')).toEqual(true);
+    });
+};
+
+const checkForbiddenNoCsrf = path => request(setUpFakeApp())
+  .post(path)
+  .set('Cookie', [mockAuthorisedCookie])
+  .type('form')
+  .send({})
+  .then((res) => {
+    expect(res.status).toEqual(403);
+  });
+
+
+const checkRedirectToLogin = async (csrfPagePath, postPath) => {
+  const { cookies, csrfToken } = await getCsrfTokenFromGet(
+    setUpFakeApp(), csrfPagePath, mockAuthorisedCookie,
+  );
+
+  return request(setUpFakeApp())
+    .post(postPath)
+    .type('form')
+    .set('Cookie', [cookies])
+    .send({
+      _csrf: csrfToken,
+    })
+    .expect(302)
+    .then((res) => {
+      expect(res.redirect).toEqual(true);
+      expect(res.headers.location).toEqual('http://identity-server/login');
+    });
+};
+
+const checkLoggedInNotAuthorised = async (csrfPagePath, postPath) => {
+  const { cookies, csrfToken } = await getCsrfTokenFromGet(
+    setUpFakeApp(), csrfPagePath, mockAuthorisedCookie,
+  );
+
+  const mockUnauthorisedJwtPayload = JSON.stringify({
+    id: '88421113', name: 'Cool Dude',
+  });
+  const mockUnauthorisedCookie = `fakeToken=${mockUnauthorisedJwtPayload}`;
+
+  return request(setUpFakeApp())
+    .post(postPath)
+    .type('form')
+    .set('Cookie', [cookies, mockUnauthorisedCookie])
+    .send({ _csrf: csrfToken })
+    .expect(200)
+    .then((res) => {
+      expect(res.text.includes('data-test-id="error-page-title"')).toEqual(true);
+      expect(res.text.includes('Not authorised')).toEqual(true);
+    });
+};
```

Comment:
🙄 have we got these routes test helper functions in 2 places now?

Reply:
yeah I will tidy them up tomorrow when doing the rest of the tidy up